code/game/jobs/job_controller.dm

```diff
@@ -355,19 +355,19 @@ var/global/datum/controller/occupations/job_master
 
 		Debug("DO, Running AC2")
 
-		// For those who wanted to be civilians if their preferences were filled, here you go.
-		for(var/mob/new_player/player in unassigned)
-			if(player.client.prefs.alternate_option == BE_ASSISTANT)
-				Debug("AC2 Assistant located, Player: [player]")
-				AssignRole(player, "Civilian")
-
+		// Antags, who have to get in, come first
 		for(var/mob/new_player/player in unassigned)
 			if(player.mind.special_role)
 				GiveRandomJob(player)
+				if(player in unassigned)
+					AssignRole(player, "Civilian")

```

Comment:
First, do a loop through all unassigned players - if they've been awarded a special role, give them a random job - or civilian, if no position is open.